src/beacon-api/app/services/scheduler_service.py

```
@@ -167,265 +278,189 @@ async def compute_and_store_performance(
+            async with ts_semaphore:
+                return await fetch_feeds_for_device(
+                    info["device_number"], d_name, token,
+                    read_key=info.get("read_key"),
+                    category=info.get("category", "lowcost"),
+                    start_date=start_dt_str, end_date=end_dt_str,
+                )
+
+        ts_devices = [d for d in device_names if device_info[d].get("device_number") is not None]
+        ts_tasks = [_fetch_ts(d) for d in ts_devices]
+        ts_results = await asyncio.gather(*ts_tasks, return_exceptions=True)
+        for d_name, res in zip(ts_devices, ts_results):
```

Ruff (0.15.9)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)